### PR TITLE
Move more trigger tests to scala-test

### PR DESCRIPTION
**Pull request by cocreature**
This PR moves more tests of triggers over to the Scala test suite, in
particular:

- The existing tests there abstract over the time mode and are
  instantiated once for wallclock mode and once for static time mode.
- I’ve added tests for TLS and Auth.
- I’ve removed the TLS and Auth tests outside of scala-test since they
  are now redundant.
- I’ve added the time tests to the scala-tests stuff since with the
  new ledger time model, that’s necessary to actually trigger a
  failure if you get static time vs wallclock time wrong (MRT and LET
  no longer exist).

I haven’t yet moved all the func tests over, I’ll do that separately
and then we can kill the old tests completely.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: triggers/tests/BUILD.bazel [hunk 1]
```diff
@@ -108,16 +106,23 @@ client_server_test(
     server_files = ["$(rootpath :acs.dar)"],
 )
 
-da_scala_test_suite(
+da_scala_test(
```
Comment (cocreature) [inline triggers/tests/BUILD.bazel:109]: They depend on each other so `scala_test_suite` doesn’t work. Also they run in less than 15 seconds on my machine so it doesn’t seem that useful to parallelize them.

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  this.getClass.getSimpleName can {
+    "AcsTests" should {
+      val assetId = LedgerApi.Identifier(packageId, "ACS", "Asset")
+      val assetMirrorId = LedgerApi.Identifier(packageId, "ACS", "AssetMirror")
+      def asset(party: String): CreateCommand =
+        CreateCommand(
+          templateId = Some(assetId),
+          createArguments = Some(
+            LedgerApi.Record(fields =
+              Seq(LedgerApi.RecordField("issuer", Some(LedgerApi.Value().withParty(party)))))))
```
Comment (cocreature) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala:39]: I currently duplicate these 7 lines for the tls and auth tests. Couldn’t quite find a sensible location  to put them an dfor now this doesn’t seem like an issue.

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractTriggerTest.scala [hunk 3]
```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.engine.trigger.test
```
Comment (aherrmann-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractTriggerTest.scala:4]: Why not pull this out into a separate `scala_library` for testing utils and have the actual tests stay in a `scala_test_suite`? If not for parallelization, it also just seems cleaner to separate those concerns.

Comment (cocreature) [reply triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractTriggerTest.scala:4]: I’ve pulled it into a `test-utils` library now. Parallelizing it made the tests run slower when run in parallel (presumably JVM overhead) so I’ve kept it in a single test suite for now.

File: triggers/tests/BUILD.bazel [hunk 4]
```diff
@@ -108,16 +106,23 @@ client_server_test(
...
-    data = [":acs.dar"],
+    srcs =
+        [
+            "src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala",
+            "src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractTriggerTest.scala",
+            "src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/FuncTestsStaticTime.scala",
+            "src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/FuncTestsWallClock.scala",
+            "src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/Jwt.scala",
+            "src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/Tls.scala",
+        ],
```
[the remark this answers is not in the document]
Comment (cocreature) [reply triggers/tests/BUILD.bazel]: That’s what I get for assuming buildifier is going to clean up my mess …

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractTriggerTest.scala (hunk 3 above)
Comment (aherrmann-da) [reply triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractTriggerTest.scala:4]: Thanks!

**Pull request merged**